subprojects/docs/src/samples/eclipse/kotlin/build.gradle.kts

```diff
@@ -0,0 +1,76 @@
+// tag::module-when-merged[]
+import org.gradle.plugins.ide.eclipse.model.AbstractClasspathEntry
+// end::module-when-merged[]
+// tag::module-before-merged[]
+// tag::module-when-merged[]
+import org.gradle.plugins.ide.eclipse.model.Classpath
+// end::module-when-merged[]
+// end::module-before-merged[]
+// tag::project-before-merged[]
+import org.gradle.plugins.ide.eclipse.model.Project
+// end::project-before-merged[]
+// tag::wtp-with-xml[]
+import org.w3c.dom.Element
+// end::wtp-with-xml[]
+
+// tag::use-eclipse-plugin[]
+// tag::use-eclipse-wtp-plugin[]
+plugins {
+    // end::use-eclipse-plugin[]
+// end::use-eclipse-wtp-plugin[]
+    war
+// tag::use-eclipse-plugin[]
+    eclipse
+// end::use-eclipse-plugin[]
+// tag::use-eclipse-wtp-plugin[]
+    `eclipse-wtp`
+// end::use-eclipse-wtp-plugin[]
+// tag::use-eclipse-plugin[]
+// tag::use-eclipse-wtp-plugin[]
+}
+// end::use-eclipse-plugin[]
+// end::use-eclipse-wtp-plugin[]
+
+
+// tag::module-before-merged[]
+// tag::module-when-merged[]
+
+eclipse.classpath.file {
+// end::module-when-merged[]
+    beforeMerged(Action<Classpath> {
+        entries.removeAll { entry -> entry.kind == "lib" || entry.kind == "var" }
+    })
+// end::module-before-merged[]
+// tag::module-when-merged[]
+    whenMerged(Action<Classpath> { ->
+        entries.filter { entry -> entry.kind == "lib" }
+            .forEach { (it as AbstractClasspathEntry).isExported = false }
```

Comment:
`AbstractClasspathEntry` is public but requiring a cast is annoying.

@donat, couldn't the `exported` boolean property be exposed by `Classpath`?

Reply:
The `Output` concrete class doesn't have an `exported` property, that's why you have to cast there. The problem is: the code is part of the public API, so I wouldn't bother changing it only for making it cleaner.

Reply:
Thanks @donat 
Let's keep this as is for now